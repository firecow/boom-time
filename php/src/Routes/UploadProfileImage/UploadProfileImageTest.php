<?
declare(strict_types=1);

namespace App\Routes\UploadProfileImage;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class UploadProfileImageTest extends RegressionTest
{

    /**
     * @return string
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPAuthorization("Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTIyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6Ijc0NTA3NTFmMzA3YmM3NmQ5NjBmMmQ3OTJjNjZmNzIwIiwiaW1nIjoiaHR0cDovL2ZpcmVjb3cuZGsvaW1hZ2VzL2JnLnBuZyIsInVuYSI6IkZpcmVjb3cifQ.uG03s_iBmB2Zmd3ZBf-MAnlmyWPRSPsUT9FLoH7CSGk");
        $superGlobals->setHTTPFilesData("image", array(
            "name" => "UploadProfileImageTest.png",
            "type" => "image\/png",
            "tmp_name" => "src/Routes/UploadProfileImage/UploadProfileImageTest.png",
            "error" => 0,
            "size" => 69376
        ));
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/uploadprofileimage/");
        return $routeResponse->prettyPrint();
    }
}