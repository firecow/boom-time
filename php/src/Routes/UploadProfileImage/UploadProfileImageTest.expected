HTTP/1.1 200 Ok
Content-Length: 38
Content-Type: application/json

{
    "photoId": "000000063064390006000006"
}

+ {
+ 	"_id": {
+ 		"$oid": "000000063064390006000006"
+ 	},
+ 	"mime": "image/png",
+ 	"width": 16,
+ 	"height": 16,
+ 	"uploaded": {
+ 		"$date": "2018-02-22T13:02:02.000Z"
+ 	},
+ 	"lastSeen": null,
+ 	"seenCount": 0,
+ 	"blob": {
+ 		"$binary": "iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH4gcGBgkM3OoYNQAAAtZJREFUOMtFk8trnGUUh5/zfu83l28ukSaToQihKF2ouLbWRSgIBQtFUDcKoiJ4W0iXStGFm7rxL9CVKxeCIJGKeFvYhReIYhUbSZqMlplMM5lkvrm93/ue18VoPXDg+cFZHX6PnDuDXL784tm1tbUn0pIpgUBkMQIiEP/LEZBI4dTt7e1+/NbbH1yzZx4+tXZiafBFeylUtcgRI8QYSa3Q7Ue6B4a72xWWl8ZEERCgWmfUPHppff3Ufba3P13R+V61GANRiRpRqfHhBpy/8CQHow6txk/cco4w9awue8Aggazbm64Y71WIASIEJ4AhS2fc2I6cfuA2RQFJEth4z9IAghckAQ2BwquxIGgwaGGIouwPUsxICFoFfmE6y/n8Dc9TlcBxVqGdTqCAqIIg2KjHaLFH1CYaauSTKtvvw13DnMGwhpsELo48g7khvVLwTVKi9VxGw94EZtgYZ4Rihhv1QeBkrcwPhzXkOmzemCDO0escUnaea92U1Tcdp7MBuwNQD1YEJELwlp+vG7Y2SuTfDbnHK59+NCf+NefBrucQaK6UWXaemUswNiAJmBhZgIn83Wvy52ZBGiIC3P6k4MKzT/N1xeAAc2DZ+ixj5MuogxjAEsFPwVcDj507oEKF7pUEr57GM5Y/8qvcO1e+ffwEL7w6JDjFALkuimUBxIIYyIfw65cJoVXQW82QdyZ0mbMKZL8d4bHE4DAJqC6aatVDMQGtA0mZ9kPC2dcM43Gdr16ecfJSmWFumG0J3a6nlYFa0AKiglVdQJFDksy5uD5HYqRipvx4f8rzjzgiAf8omAjYxb2GxdrhqOQ7+9BIHKIRky6eE3XG669kOD3CjSHNoHBQBPAKnX6JwyO83dlLd7//PXl351Y4byQk6sGUgFgg4YjNbVD3v51JAmqS0B9Uru50wk0BaLfqzXpt1lb1yR117+i7EDD+y0RIrA35uNLr9fPjfwAcTICPfAiVbgAAAABJRU5ErkJggg==",
+ 		"$type": "00"
+ 	}
+ }
